src/core/qgsabstractdatabaseproviderconnection.h

```diff
@@ -0,0 +1,406 @@
+        //! Name of the geometry column
+        QString                       mGeometryColumn;
+        //! The number of geometry columns in the table
+        int                           mGeometryColumnCount;
+        //! PK columns
+        QStringList                   mPkColumns;
+        TableFlags                    mFlags;
+        QString                       mComment;
+        //! SQL where condition
+        QString                       mSql;
+        //! Additional unstructured information about the table
+        QMap<QVariant, QVariant>      mInfo;
+    };
+
+
+    /**
+     * The Capability enum represent the operations supported by the connection
+     */
+    enum Capability
+    {
+      CreateVectorTable = 1 << 1,   //!< Can CREATE a vector (or aspatial) table/layer
+      // CreateRasterTable does not make much sense, skip it for now:
+      DropRasterTable = 1 << 2,     //!< Can DROP a raster table/layer
+      DropVectorTable = 1 << 3,     //!< Can DROP a vector (or aspatial) table/layer
+      RenameTable = 1 << 4,         //!< Can RENAME a table/layer
+      CreateSchema = 1 << 5,        //!< Can CREATE a schema
+      DropSchema = 1 << 6,          //!< Can DROP a schema
+      RenameSchema = 1 << 7,        //!< Can RENAME a schema
+      ExecuteSql = 1 << 8,          //!< Can execute raw SQL queries (without returning results)
+      // TODO: Transaction = 1 << 9,   //!< Supports transactions when executing operations
+      Vacuum = 1 << 10,             //!< Can run vacuum
+      Tables = 1 << 11,             //!< Can list tables
+      Schemas = 1 << 12,            //!< Can list schemas (if not set, the connection does not support schemas)
+      SqlLayers = 1 << 13,          //!< Can create vector layers from SQL SELECT queries
+      // TODO: TruncateTable = 1 << 14,      //!< Can TRUNCATE a table
+      TableExists = 1 << 15,        //!< Can check if table exists
+      Spatial = 1 << 16,            //!< The connection supports spatial tables
+    };
+
+    Q_ENUM( Capability )
+    Q_DECLARE_FLAGS( Capabilities, Capability )
+    Q_FLAG( Capabilities )
+
+    /**
+     * Creates a new connection with \a name by reading its configuration from the settings.
+     * If a connection with this name cannot be found, an empty connection will be returned.
+     */
+    QgsAbstractDatabaseProviderConnection( const QString &name );
+
+    /**
+     * Creates a new connection with \a name and initializes the connection from the \a uri.
+     * The connection is not automatically stored in the settings.
+     * \see store()
+     */
+
+    QgsAbstractDatabaseProviderConnection( const QString &name, const QString &uri );
+
+    // Public interface
+
+    /**
+     * Returns connection capabilities
+     */
+    Capabilities capabilities() const;
+
+    // Operations interface
+
+    /**
+     * Creates an empty table with \a name in the given \a schema (schema is ignored  if not supported by the backend)
```

Comment:
* Raises a  QgsProviderConnectionException if any errors are encountered 

(for all these methods)